### ISSUE TITLE
Server doesn't appear to be fully 8bit clean

### ISSUE THREAD

**Issue by gutschke** [andris9/smtp-server#39]
While some UTF8-encoded characters make it through just fine, others appear to be mangled.

I can open a connection to my server and send things such as "äöü€" in the message body just fine. But if I want to send characters from the Latin Extended Additional page (e.g. "ẅ"), it gets corrupted. As far as I can tell, there is no way for my code to ever retrieve the original character. It inevitably gets turned into � (U+FFFD). I believe that means that Javascript tried to decode the input and failed.

But if I understand correctly, "ẅ" has been part of Unicode for a long time and should be supported by Javascript. It is U+1E8B. I am a little puzzled where things go wrong, but maybe you could consider a option to turn the server into "binary" mode. Don't even try to interpret data as UTF-8. Just pass it along as an uninterpreted byte stream. That would presumably avoid this and all similar problems.

**Comment by andris9** [comment]
Where do you have problems with utf8? With envelope addresses? MIME message? Message contents are passed to your app as binary Buffer, you get the exact bytes that the client sent you, there is no string conversion in between. Unicode envelope addresses should work just fine.

**Comment by gutschke** [comment]
I sincerely apologize for the bogus bug report.

I was testing all sorts of crazy corner cases for the server, and I used "telnet -8" to manually talk to it. Turns out "telnet" isn't quite as transparent as I thought it would be, even when given the "-8" option. It apparently does some form of Unicode validation and, unfortunately, gets it wrong. Oops.

After reading your comment, I devised a different test and I can now successfully send even "strange" binary characters to my smtp-server. Again, sorry for the false alarm; I should have questioned by testing methodology before blaming your code.
